core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -168,6 +178,13 @@ private[spark] abstract class YarnSchedulerBackend(
         logInfo(s"ApplicationMaster registered as $am")
         amEndpoint = Option(am)
 
+      case ReregisterClusterManager(am) =>
```

Comment:
Instead of a new message, why not just re-use the existing `RegisterClusterManager` one? If the amEndpoint is currently set, then call `reset`.